### PR TITLE
Change `TimestampGen` unit in integration test from millisecond to microsecond

### PR DESCRIPTION
**Pull request by ttnghia**
In integration test, class `TimestampGen` generate timestamps that are different in milliseconds as the smallest unit. This PR changes that to microseconds, which allows testing at a lower time unit.

Depends on:
 * https://github.com/NVIDIA/spark-rapids/pull/6311

### CODE REVIEW

**Review by andygrove — APPROVED** [review]

**Comment by ttnghia** [comment]
build

**Comment by ttnghia** [comment]
build

**Review by revans2 — APPROVED** [review]

**Comment by ttnghia** [comment]
CI failed due to another bug in casting array to string (test failed: `test_cast_array_to_string`).

**Comment by ttnghia** [comment]
I filed a related issue: https://github.com/NVIDIA/spark-rapids/issues/6294

**Comment by ttnghia** [comment]
build

**Comment by ttnghia** [comment]
CI failed again due to a new bug. I filed a related issue: https://github.com/NVIDIA/spark-rapids/issues/6302

**Comment by ttnghia** [comment]
So this PR has opened a can of bugs that all need to be fixed. Should I xfail all the failed tests and merge this first, then fix these bugs later in separate PRs? It seems that fixing these bugs requires the changes in the integration test in this PR, but this PR can't be merged before these bugs are fixed, and fixing these bugs in this PR would be too off-topic.

**Comment by ttnghia** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala [hunk 1]
```diff
@@ -176,7 +176,7 @@ object GpuCast extends Arm {
    */
   private val TIMESTAMP_TRUNCATE_REGEX = "^([0-9]{4}-[0-9]{2}-[0-9]{2} " +
     "[0-9]{2}:[0-9]{2}:[0-9]{2})" +
-    "(.[0-9]*(?:0)?[1-9]+)?(.0*[1-9]+)?(?:.0*)?$"
+    "(.0*[1-9]*(?:0)?[1-9]+)?(.0*[1-9]+)?(?:.0*)?$"
```
Comment (andygrove) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala:179]: I don't think this works in all cases. I am working on a fix for this.

Comment (andygrove) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala:179]: See https://github.com/NVIDIA/spark-rapids/pull/6311

**Comment by ttnghia** [comment]
New failed tests related to ORC writing.

**Comment by ttnghia** [comment]
build

**Comment by ttnghia** [comment]
New failed test again, but this time is due to an old issue (parquet writing).

**Comment by ttnghia** [comment]
build

**Review by revans2 — APPROVED** [review]
Looks good

**Pull request merged**